helix-term/src/ui/editor.rs

```diff
@@ -276,29 +292,32 @@ impl EditorView {
             )
         };
 
-        // basically just queries the theme color defined in the config
-        let hint = get_scope_of("diagnostic.hint");
-        let info = get_scope_of("diagnostic.info");
-        let warning = get_scope_of("diagnostic.warning");
-        let error = get_scope_of("diagnostic.error");
-        let r#default = get_scope_of("diagnostic"); // this is a bit redundant but should be fine
+        let highlight = match severity {
+            Some(Severity::Error) => get_scope_of("diagnostic.error"),
```

Comment:
With this refactor we don't need `get_scope_of` to be a separate closure anymore

```rust
let scope = match severity { .. };
let highlight = theme.find_scope_index(scope).or_else(// ...
```

Reply:
fixed in c3f652f